Update: Improve frequency of log data and information recorded

danielghost opened the pull request:
Fixes #25.

### Update
* Consolidate logs into one entry per set on `update`.
* Only record data when changed from the previous entry.
*  Include context for how the set was modified for easier debugging and better understanding of the user journey.
* Added methods for retrieving model specific scores from a set.

### Dependency
* https://github.com/adaptlearning/adapt-contrib-scoring/pull/21.

danielghost committed:
Update: Consolidate logs into one entry per set on `update` and include context for how the set was modified for easier debugging and better understanding of the user journey (fixes #25).

oliverfoster commented:
Could you merge in master please?

danielghost committed:
Merge branch 'master' into issue/25

danielghost commented on js/ScoringSet.js:
```diff
@@ -413,6 +467,97 @@ export default class ScoringSet extends Backbone.Controller {
+        minScore: isAvailable ? minScore : -minScore,
+        maxScore: isAvailable ? maxScore : -maxScore
+      };
+      if (questionModel.get('_isSubmitted')) data.score = -this.getScoreByModel(questionModel);
```
Think this needs looking into to account for both available and unavailable. Currently expecting unavailable only.
``` JavaScript
const score = this.getScoreByModel(questionModel);
if (questionModel.get('_isSubmitted')) data.score = isAvailable ? score : -score;
```

oliverfoster replied:
Cool. I think just commit as is, and then look at the problem again in the new version. Just make some notes, possibly on an New issue, about what the problem is so that you can test that it's been fixed or nullified later?

danielghost committed:
Update log score accordingly based on availability. Only include question modifiers that are within the set.

cahirodoherty-learningpool reviewed (approved):
👀

joe-replin committed:
Change lifecycle events to log as info()

Keeping the state change events as debug(). This provides clearer visibility of key events while troubleshooting.
